client/src/app/site/projector/services/current-list-of-speakers.service.ts

```diff
@@ -0,0 +1,124 @@
+import { Injectable } from '@angular/core';
+
+import { BehaviorSubject, Observable, Subscription } from 'rxjs';
+
+import { ProjectorService } from 'app/core/core-services/projector.service';
+import { ProjectorRepositoryService } from 'app/core/repositories/projector/projector-repository.service';
+import { ViewListOfSpeakers } from 'app/site/agenda/models/view-list-of-speakers';
+import { isBaseViewModelWithListOfSpeakers } from 'app/site/base/base-view-model-with-list-of-speakers';
+import { SlideManager } from 'app/slides/services/slide-manager.service';
+import { ViewProjector } from '../models/view-projector';
+import { ListOfSpeakersRepositoryService } from 'app/core/repositories/agenda/list-of-speakers-repository.service';
+
+/**
+ * Observes the projector config for a given projector and returns a observable of the
+ * current view list of speakers displayed on the projector.
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class CurrentListOfSpeakersService {
+    /**
+     * This map holds the current (number or null) los-id for the the projector.
+     * It is used to check, if the reference has changed (this clos id changed for one projector).
+     */
+    private currentListOfSpeakersIds: { [projectorId: number]: number | null } = {};
+
+    /**
+     * Active subscriptions for the clos of each projector.
+     */
+    private currentListOfSpeakersSubscriptions: { [projectorId: number]: Subscription } = {};
+
+    /**
+     * All subjects for all clos of each projector.
+     */
+    private currentListOfSpeakers: { [projectorId: number]: BehaviorSubject<ViewListOfSpeakers | null> } = {};
```

Comment:
please add a public getter for an observable to the current list of speakers
`public get currentListOfSpeakers(): Observable<ViewListOfSpeakers>`

no reference id, no projector

Reply:
We do have just one reference projector, right?

I guess the code comes from the time with multiple ones ;)

Reply:
there was an idea to add multiple reference projectors and multiple list of speakers. It was later discarded

Reply:
Not only the idea, but we *had* this already implemented what then was discarded (except for this service) ;)